setup.py

```diff
@@ -2,18 +2,22 @@
 import os
 import sys
 
+from packaging.version import parse
 from setuptools import find_packages, setup
 
-min_version, max_version = ((3, 9), "3.9"), ((3, 11), "3.11")
-if not (min_version[0] <= sys.version_info[:2] <= max_version[0]):
-    # Python 3.5 does not support f-strings
+supported_python_versions = ["3.9", "3.10", "3.11"]
```

Comment:
this isn't strictly necessary but seems easier include only the version strings (rather than the string and the parts tuple like before). 

It also requires we specify all supported versions rather than a min and max because it made it easier to  build the matrix from it - but I'm sure I can figure it out if we truly want a range. The tricky part is what to do if we ever suport different major versions, e.g. 3.11, 3.12, and 4.1. (i.e., I wasn't sure if you said min is 3.11 and max is 4.1, how to know how many versions are between that)